Checkpoints default values for RunConfig

Immexxx opened the pull request:
With save_checkpoints_secs having a default value (of 600), the call to RunConfig fails if only save_checkpoints_steps is passed. To address this, the value is defaulted only if both the parameters are None. In this new way, the call to RunConfig will not fail if only save_checkpoints_steps is passed

Immexxx committed:
Default value of save_checkpoints_secs

To not use the default value of save_checkpoints_secs if save_checkpoints_steps is specified in the RunConfig call

Immexxx committed:
Update run_config.py

tensorflow-jenkins commented:
Can one of the admins verify this patch?

ispirmustafa reviewed (commented):
If both of them None, that means user disables the saving. This change prevents that. Instead you can use a constant _USE_DEFAULT. if it's _USE_DEFAULT and not steps then set it to 600. if it's _USE_DEFAULT and steps then set it to None.

ispirmustafa reviewed (changes requested):
If both of them None, that means user disables the saving. This change prevents that. Instead you can use a constant _USE_DEFAULT. if it's _USE_DEFAULT and not steps then set it to 600. if it's _USE_DEFAULT and steps then set it to None.

Immexxx committed:
Addressing None, None caller input 

As mentioned in the review comments, the caller many not want any checkpoints and can pass None, None for these parameters. This is to address that case.

ispirmustafa reviewed (approved):
Thanks

Immexxx commented:
@ispirmustafa  - Thanks for the review

martinwicke commented:
Jenkins, test this please.

martinwicke committed:
Fix indentation

martinwicke commented:
Jenkins, test this please.

martinwicke commented:
Please fix the test failures. `_USE_DEFAULT` must be qualified (as `RunConfig._USE_DEFAULT`).

Immexxx commented:
@martinwicke Thanks. Will do

Immexxx committed:
Update run_config.py

martinwicke commented:
Jenkins, test this please.
